src/core/core.controller.js

```diff
@@ -423,41 +423,47 @@ module.exports = function(Chart) {
 			return elementsArray;
 		},
 
-        getElementsAtXAxis: function(e){
-            var me = this;
-            var eventPosition = helpers.getRelativePosition(e, me.chart);
-            var elementsArray = [];
-
-            var found = (function() {
-                if (me.data.datasets) {
-                    for (var i = 0; i < me.data.datasets.length; i++) {
-                        var meta = me.getDatasetMeta(i);
-                        if (me.isDatasetVisible(i)) {
-                            for (var j = 0; j < meta.data.length; j++) {
-                                if (meta.data[j].inLabelRange(eventPosition.x, eventPosition.y)) {
-                                    return meta.data[j];
-                                }
-                            }
-                        }
-                    }
-                }
-            }).call(me);
-
-            if (!found) {
-                return elementsArray;
-            }
-
-            helpers.each(me.data.datasets, function(dataset, datasetIndex) {
-                if (me.isDatasetVisible(datasetIndex)) {
-                    var meta = me.getDatasetMeta(datasetIndex);
-                    if(!meta.data[found._index]._view.skip){
-                    	elementsArray.push(meta.data[found._index]);
-                    }
-                }
-            }, me);
-
-            return elementsArray;
-        },		
+		getElementsAtXAxis: function(e) {
+			var me = this;
+			var eventPosition = helpers.getRelativePosition(e, me.chart);
+			var elementsArray = [];
+
+			var found = (function() {
+				if (me.data.datasets) {
+					for (var i = 0; i < me.data.datasets.length; i++) {
+						var meta = me.getDatasetMeta(i);
+						if (me.isDatasetVisible(i)) {
+							for (var j = 0; j < meta.data.length; j++) {
+								if (meta.data[j].inLabelRange(eventPosition.x, eventPosition.y)) {
+									return meta.data[j];
+								}
+							}
+						}
+					}
+				}
+			}).call(me);
+
+			if (!found) {
+				return elementsArray;
+			}
+
+			helpers.each(me.data.datasets, function(dataset, datasetIndex) {
+				if (me.isDatasetVisible(datasetIndex)) {
+					var meta = me.getDatasetMeta(datasetIndex);
+					var item = null;
+					var findByXCoord = function (it) {
+						if (found._model.x === it._model.x)
+							item = it;
+					};
+					meta.data.forEach(findByXCoord);
```

Comment:
We shouldn't iterate other all items here, but break as soon as one is found. You should maybe use a for loop or the `helpers.findIndex` instead.